src/raft/replication.c

```diff
@@ -1674,6 +1674,50 @@ static void takeSnapshotClose(struct raft *r, struct raft_snapshot *s)
 	r->fsm->snapshot_finalize(r->fsm, &s->bufs, &s->n_bufs);
 }
 
+
+static unsigned dynamicTrailingIndex(struct raft *r, struct raft_snapshot *snapshot) {
+	struct raft_log *l = r->log;
+	size_t snapshot_size = 0, num_entries = logNumEntries(l);
+	unsigned trailing, i;
```

Comment:
> unsigned

I was triggered here because I started to think about the relation between number of bytes needed to represent `snapshot->n_bufs` and `i` which is used for indexing here. Could you either change `unsigned` to `size_t` or explain why it is sufficient.

Reply:
Ok fair. I used `unsigned` because `snapshot->n_bufs` is an `unsigned` and the return value is and `unsigned`, too.

However you are right, `num_entries` is a `size_t`, so we should really be consistent there to avoid weird stuff.

Given that the return value must be an `unsigned`, but we are capping the value to the settings `trailing` I think it's fair to convert after the capping has taken place as if `unsigned` is narrower than `size_t` we still know that no truncation happens.